internal/rukpak/util/hash_test.go

```diff
@@ -0,0 +1,117 @@
+package util_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/operator-framework/operator-controller/internal/rukpak/util"
+)
+
+func TestDeepHashObject(t *testing.T) {
+	tests := []struct {
+		name         string
+		wantErr      bool
+		obj          interface{}
+		expectedHash string
+	}{
+		{
+			name:    "populated obj with exported fields",
+			wantErr: false,
+			obj: struct {
+				Str string
+				Num int
+				Obj interface{}
+				Arr []int
+				B   bool
+				N   interface{}
+			}{
+				Str: "foobar",
+				Num: 22,
+				Obj: struct{ foo string }{foo: "bar"},
```

Comment:
Nit: might want to also export `Foo` here. As is, `Obj` will be encoded as `{}` since foo is still unexported.

Reply:
Oops, missed that one. Thanks!